src/test/java/integration/DiscountPerCounterpartyControllerIT.java

```diff
@@ -0,0 +1,267 @@
+package integration;
+
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.TimeZone;
+import java.util.UUID;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.opinta.dto.DiscountPerCounterpartyDto;
+import com.opinta.entity.Counterparty;
+import com.opinta.entity.Discount;
+import com.opinta.entity.DiscountPerCounterparty;
+import com.opinta.exception.IncorrectInputDataException;
+import com.opinta.mapper.DiscountPerCounterpartyMapper;
+import com.opinta.service.DiscountPerCounterpartyService;
+import integration.helper.TestHelper;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import static org.junit.Assert.assertEquals;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static java.util.TimeZone.getTimeZone;
+
+import static javax.servlet.http.HttpServletResponse.SC_BAD_REQUEST;
+import static javax.servlet.http.HttpServletResponse.SC_OK;
+
+import static io.restassured.module.mockmvc.RestAssuredMockMvc.given;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.fail;
+import static org.skyscreamer.jsonassert.JSONAssert.assertEquals;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+public class DiscountPerCounterpartyControllerIT extends BaseControllerIT {
+    private static final String ISO_8601_24H_FULL_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSXXX";
+    private static final TimeZone UTC = getTimeZone("UTC");
+
+    @Autowired
+    private DiscountPerCounterpartyService discountPerCounterpartyService;
+    @Autowired
+    private DiscountPerCounterpartyMapper discountPerCounterpartyMapper;
+    @Autowired
+    private TestHelper testHelper;
+    private List<Discount> discounts;
+    private List<DiscountPerCounterparty> discountsPerCounterparty;
+    private Counterparty counterparty;
+    private JSONParser jsonParser = new JSONParser();
+    private SimpleDateFormat simpleDateFormat;
+    private ObjectMapper jsonMapper = new ObjectMapper();
+    
+    @Before
+    public void setUp() throws Exception {
+        counterparty = testHelper.createCounterparty();
+        discounts = testHelper.createDiscounts();
+        discountsPerCounterparty = testHelper.createDiscountsPerCounterparty(discounts, counterparty);
+        TimeZone.setDefault(UTC);
+        simpleDateFormat = new SimpleDateFormat(ISO_8601_24H_FULL_FORMAT);
+        simpleDateFormat.setTimeZone(UTC);
+    }
+    
+    @After
+    public void tearDown() throws Exception {
+        testHelper.deleteDiscountsPerCounterparty(discountsPerCounterparty);
+        testHelper.deleteDiscounts(discounts);
+        testHelper.deleteCounterparty(counterparty);
```

Comment:
Plz, move delete of discount and counterparty to the method deleteDiscountPerCounterparty(...)